dydx/dydxViews/dydxViews/_v4/Trade/Margin/dydxTargetLeverageView.swift

```diff
@@ -91,26 +91,15 @@ public class dydxTargetLeverageViewModel: PlatformViewModel {
                     .wrappedViewModel
             }
 
-            let selectedItems = self.leverageOptions.compactMap {
-                Text($0.text)
-                    .themeFont(fontType: .plus, fontSize: .small)
-                    .themeColor(foreground: .textPrimary)
-                    .padding(8)
-                    .frame(minWidth: width)
-                    .themeColor(background: .layer1)
-                    .borderAndClip(style: .cornerRadius(8), borderColor: ThemeColor.SemanticColor.layer5)
-                    .wrappedViewModel
-            }
-
-            return ScrollView(.horizontal, showsIndicators: false) {
-                TabGroupModel(items: items,
-                              selectedItems: selectedItems,
-                              currentSelection: self.selectedOptionIndex,
-                              onSelectionChanged: { index in
-                    self.optionSelectedAction?(self.leverageOptions[index])
-                },
-                              spacing: spacing)
-                .createView(parentStyle: parentStyle)
+            return HStack(spacing: spacing) {
+                ForEach(items.indices, id: \.self) { index in
+                    PlatformButtonViewModel(content: items[index], type: .iconType, state: .secondary) { [weak self] in
+                        guard let option = self?.leverageOptions[index] else { return }
+                        PlatformView.hideKeyboard()
+                        self?.optionSelectedAction?(option)
+                    }
+                    .createView()
```

Comment:
trying to force the selected item was proving to be a swiftui headache so opted to not use a selectable set of items, rather a set of quick action buttons